[stripe-ui-core] Support label and placeholder in `TextField`

ccen-stripe opened the pull request:
# Summary
<!-- Simple summary of what was changed. -->
Support nullable `label` and add a `placeholder` to `TextField`

# Motivation
<!-- Why are you making this change? If it's for fixing a bug, if possible, please include a code snippet or example project that demonstrates the issue. -->


# Testing
<!-- How was the code tested? Be as specific as possible. -->
- [ ] Added tests
- [ ] Modified tests
- [x] Manually verified

# Screenshots
| Placeholder without label  | Place holder selected | place holder and label |
| ------------- | ------------- | ------------- |
| ![PlaceHolder](https://user-images.githubusercontent.com/79880926/217696442-9e374efb-3838-4eaf-8398-a3137317b930.png)  | ![PlaceHolderSelected](https://user-images.githubusercontent.com/79880926/217696447-b2b4ff49-891f-47e0-aa76-7108eb5a5e20.png) | ![PlaceHolderAndLabel](https://user-images.githubusercontent.com/79880926/217696445-87a97a7d-e97b-4e05-8042-a0b75c909363.png) |




# Changelog
<!-- Is this a notable change that affects users? If so, add a line to `CHANGELOG.md` and prefix the line with one of the following:
    - [Added] for new features.
    - [Changed] for changes in existing functionality.
    - [Deprecated] for soon-to-be removed features.
    - [Removed] for now removed features.
    - [Fixed] for any bug fixes.
    - [Security] in case of vulnerabilities.
-->

carlosmuvi-stripe commented on payments-ui-core/api/payments-ui-core.api:
```diff
@@ -169,6 +169,7 @@ public abstract class com/stripe/android/ui/core/elements/CardNumberController :
 	public abstract fun getCardBrandFlow ()Lkotlinx/coroutines/flow/Flow;
 	public abstract fun getCardScanEnabled ()Z
 	public fun getEnabled ()Z
+	public fun getPlaceHolder ()Lkotlinx/coroutines/flow/Flow;
```
Why is CardNumberController exposed?

ccen-stripe replied:
it should not, marked it as internal

carlosmuvi-stripe commented on stripe-ui-core/src/main/java/com/stripe/android/uicore/elements/TextFieldController.kt:
```diff
@@ -105,7 +106,8 @@ class SimpleTextFieldController constructor(
 
-    override val label: Flow<Int> = MutableStateFlow(textFieldConfig.label)
+    override val label = MutableStateFlow(textFieldConfig.label)
+    override val placeHolder: Flow<String?> = MutableStateFlow(textFieldConfig.placeHolder)
```
Does this need to be a flow? Are there scenarios where we want placeHolder to be dynamic?

ccen-stripe replied:
Hmm for the usecase we need in identity it doesn't, but just added it to align with other fields which are all defined as flow here

jaynewstrom-stripe reviewed (commented):
I didn't see any tests. Can we add a few?

ccen-stripe commented:
> I didn't see any tests. Can we add a few?

Sure, renamed the test `TextFieldControllerTest` to `TextFieldControllerTest` and moved it from `payment-ui-core` to `stripe-ui-core`

jaynewstrom-stripe commented on stripe-ui-core/src/test/java/com/stripe/android/uicore/elements/SimpleTextFieldControllerTest.kt:
```diff
@@ -225,8 +221,30 @@ internal class TextFieldControllerTest {
         verify(config).filter("1a2b3c4d")
     }
 
+    fun `Verify null label`() {
```
I think you need to add `@Test` to these.

ccen-stripe commented on payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/CardNumberController.kt:
```diff
@@ -26,7 +26,7 @@ import kotlinx.coroutines.flow.combine
 import kotlin.coroutines.CoroutineContext
 
-sealed class CardNumberController : TextFieldController, SectionFieldErrorController {
+internal sealed class CardNumberController : TextFieldController, SectionFieldErrorController {
```
this class is incorrectly marked as public and would expose getPlaceHolder in `.api` with this PR, change it to internal
